pyccl/errors.py

```diff
@@ -1,6 +1,50 @@
-__all__ = ("CCLError", "CCLWarning", "CCLDeprecationWarning",)
+__all__ = ("CCLError", "CCLWarning", "CCLDeprecationWarning",
+           "warnings", "update_warning_verbosity")
 
-import warnings
+import warnings as warnings_builtin
+
+
+_warning_importance = {'high': 10, 'low': 1}
+_verbosity_thresholds = {'none': 100, 'low': 10, 'high': 1}
+
+
+class warnings:
+    _CCL_WARN_THRESHOLD = 10  # Equivalent to "low" verbosity
+
+    def warn(*args, **kwargs):
+        category = kwargs.get('category')
+        importance = _warning_importance[kwargs.pop('importance', 'low')]
+
+        if category == CCLWarning:
+            if importance < warnings._CCL_WARN_THRESHOLD:
+                return
```

Comment:
Just a minor thing which can cause breakage in the future if not treated right.

1. No need to double-nest.
2. Type equivalence check with `is`, otherwise it's unsafe.
```python
if category is CCLWarning and importance < warnings._CCL_WARN_THRESHOLD:
    return
```